modules/compiler/spirv-ll/source/builder_core.cpp

```diff
@@ -1949,6 +1944,17 @@ cargo::optional<Error> Builder::create<OpFunction>(const OpFunction *op) {
 
   SPIRV_LL_ASSERT_PTR(function);
 
+  if (op->FunctionControl() & spv::FunctionControlOptNoneINTELMask) {
+    SPIRV_LL_ASSERT(
+            module.isExtensionEnabled("SPV_INTEL_optnone"),
+            "SPV_INTEL_optnone must be enabled.");
+    if (module.isExtensionEnabled("SPV_INTEL_optnone")) {
```

Comment:
Can probably get rid of this `if` given we've asserted above.

Reply:
I think you're looking at my previous commit. I'll rebase and clean up.